cobigen-maven/cobigen-maven-plugin/src/main/java/com/devonfw/cobigen/maven/GenerateMojo.java

```diff
@@ -223,18 +222,26 @@ private CobiGen createCobiGenInstance() throws MojoExecutionException, MojoFailu
             }
         }
 
-        URL contextXmlUrl = classRealm.getResource("context.xml");
-        if (contextXmlUrl == null) {
-            getLog().error("No context.xml could be found in the classpath!");
-            return null;
+        Path rootPath;
+        URL rootUrl = classRealm.getResource("context.xml");
+        if (rootUrl == null) {
+            rootUrl = classRealm.getResource("src/main/templates/context.xml");
+            if (rootUrl == null) {
+                getLog().error("No context.xml could be found in the classpath!");
+                return null;
+            } else {
+                rootPath = Paths.get(URI.create(rootUrl.toString())).getParent().getParent().getParent();
+            }
+        } else {
+            rootPath = Paths.get(URI.create(rootUrl.toString()));
         }
-        getLog().debug("Found context.xml @ " + contextXmlUrl.toString());
+        getLog().debug("Found context.xml @ " + rootUrl.toString());
         final List<String> foundClasses = new LinkedList<>();
-        if (contextXmlUrl.toString().startsWith("jar")) {
-            getLog().info("Processing configuration archive " + contextXmlUrl.toString());
+        if (rootUrl.toString().startsWith("jar")) {
```

Comment:
its hard to read.
`rootUrl` seems to be the context.xml location, why not naming like this?
`rootPath` is however having different intentions. If the templates are in `src/main/templates` it is pointing to the top level folder root of the configuration. If the templates are located in the root, `rootPath` seems to point at the context.xml Oo

Reply:
Changed the names for easier reading

For clarification:
rootPath should always point at the template project root, for example `.../cobigen-templates-parent/templates-oasp4j`
Upon initilizaiton, if the context configuration is in the rootpackage, it will indeed point to context.xml first, though later on rootPath.getParent() is called to get the actual folder instead. This call is made at this point, as the other way, if the context configuration is in the src/main/templates, you have to somehow get back up 3 folders and getParent() is called thrice on rootPath, after which you have `.../cobigen-templates-parent/templates-oasp4j/src` as you are also starting from the context.xml itself. 
That way, in both occasions, after the next call of getParent() in line 282 you'll get the project root.

If the given input is a jar, the root package is gotten in a different way altogether, splitting the path at the location of the jar indicator `!`, which will be present in both cases too but should not have to be called anyway as the template jar should not have a src/main/templates if i remember correctly

Reply:
Still not done in an elegant way but the easiest way i found to do it without code redundancy.